eg/Classes/PluginInstall.py

```diff
@@ -146,38 +243,78 @@ def Import(self, filepath):
                 if os.path.isdir(path):
                     basePath = path
                     break
-            dialog = PluginOverviewDialog(
-                title="Install EventGhost Plugin",
-                pluginData=pluginData,
-                basePath=basePath,
-                message=(
-                    "Do you really want to install this plugin into "
-                    "EventGhost?"
-                )
-            )
-            result = dialog.ShowModal()
-            dialog.Destroy()
-            if result == wx.ID_CANCEL:
-                return
+
             guid = pluginData['guid']
             if guid in eg.pluginManager.database:
                 # a plugin with same GUID already exists
                 info = eg.pluginManager.database[guid]
-                if info.path.lower().startswith(eg.localPluginDir.lower()):
-                    # plugin with same GUID exists in user dir, so delete
-                    # the folder first
+                name = info.englishName
+                mData = (name, pluginData['name'])
+                if info.englishName != pluginData['name']:
+                    title = Text.replaceTitle % info.englishName
+                    message = Text.replaceMessage % mData
+
+                elif info.version != pluginData['version']:
+                    mData = (name, info.version, pluginData['version'])
+                    if info.version < pluginData['version']:
+                        title = Text.upgradeTitle % name
+                        message = Text.upgradeMessage % mData
+                    else:
+                        title = Text.downgradeTitle % name
+                        message = Text.downgradeMessage % messageData
+                else:
+                    title = Text.overwriteTitle % name
+                    message = Text.overwriteMessage % mData
+
+                dialog = PluginOverviewDialog(
+                    title=title,
+                    message=message,
+                    pluginData=pluginData,
+                    basePath=basePath,
+                )
+
+                result = dialog.ShowModal()
+                dialog.Destroy()
+                if result == wx.ID_CANCEL:
+                    return
+
+                self.BackupPlugin(info)
+                try:
                     shutil.rmtree(info.path, False)
-            dstDir = os.path.join(eg.localPluginDir, os.path.basename(basePath))
+                except WindowsError:
+                    if info.path.startswith(eg.corePluginDir):
+                        eg.PrintNotice(Text.pluginRemoveError + info.path)
+                    else:
+                        raise
```

Comment:
the reason for the reraise is because of some unknown problem. permissions error or what have you.

We do know that with windows 10 it will not allow for a program to modify anything in the program files folder. so if EG goes to try and remove the old plugin and it is located in the core plugins directory which is normally in the program files folder. there is a really good change that is why an error has occurred.. tho i do not specifically state this due to installation location and things of that nature. but by checking to see if the plugin was housed in the core plugin dir. the chances are pretty good that it has an issue removing it.. so letting the user know that they have to do it manually I thought would be the best thing..

now if the plugin is not in that folder.. then I have not a clue what the issue could have been. and reraising is what i thought would be best. originally there was no message dialog explaining any kind of an error. it would just throw an exception.

so this is one step better... 

and as far as the icon thing... i do agree with you on this.. and I think there should be a eg.PrintInfo for this purpose. and print should have no icon what's so ever. but as it is set up now print has the nice information icon. now. we can change that up if you like it's pretty simple to do. I am personally for it..
but.. i also think it should be one step further.. and there should be some kind of separation between plugin PrintError, PrintNotice and PrintInfo..    VS a core EG one. and an icon is a good way of giving that. I have thought about this in the past... but I do think that at the very least migrating the information icon away from stdout and put it into a new method in eg.Log is a good choice. this way it will also get a time stamp where as the print does not... and also i think that doing a print is really for debugging purposes and nothing more, and has no need for an icon.


and if memory serves the english name is what the guid gets set to.. I will have to check.. and the pluginName is the class name.. I think... again i will have to look

heh sorry forgot to remove the import re.